App/src/main/java/com/google/sps/servlets/ShelterMatch.java

```diff
@@ -49,41 +53,116 @@ public void addShelters(){
+    //returns a Shelter[] of the sorted results based on score
+    public static Shelter[] sortTop3(){
+        Shelter[] output = new Shelter[3];
+        ArrayList<Shelter> temp = new ArrayList<Shelter>();
 
+        //populating temp with Shelters from list so temp can be altered without effecting list
+        for(int i = 0; i < list.size(); i++){
+            temp.add(list.get(i));
         }
+        
+        int count = 0;
+        while(count <= output.length-1){
+            int posMax = 0;
+            int maxScore = temp.get(0).getScore();
 
+            for(int j = 0; j < temp.size(); j++){
+                if(maxScore < temp.get(j).getScore()){
+                    maxScore = temp.get(j).getScore();
+                    posMax = j;
+                }
+                else if(maxScore == temp.get(j).getScore()){
+                    maxScore = maxScore;
+                }
+            }
+            
+            output[count] = temp.get(posMax);
+            temp.remove(posMax);
+            count++;
+        }
+
+        //checking to see if any shelters with score of 0 was added to top3
+        for(int i = 0; i < output.length; i++){
+            if(output[i].getScore() == 0)
+                output[i] = new Shelter("No other available shelters",null,0L,null,0,0,false);
+        }
+        return output;
     }
+
+    public static boolean ageRequire(User person, int i){
+        //        
+        if(person.getAge() >= list.get(i).getAgeRange()[0] && person.getAge() <= list.get(i).getAgeRange()[1]){
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean genderRequire(User person, int i){
+        //        
+        if(person.getGender().equals(list.get(i).getGender())){
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean disability(User person, int i){
+        //        
+        if(person.getDisabilities().equals(list.get(i).getDisability())){
+            return true;
+        }
+        return false;
+    }
+
     //returns an array of the top 3 shelters matched with the user
     public static Shelter[] isEligible(User person){
         Shelter[] top3 = new Shelter[3];
-
+        
         //comparing each shelter age requirement with the user        
         for(int i = 0; i < list.size(); i++){
-            if(person.getAge() >= list.get(i).getAgeRange()[0] && person.getAge() <= list.get(i).getAgeRange()[1]){
+            if(ageRequire(person,i))
+                list.get(i).addScore();
+            if(genderRequire(person,i))
+                list.get(i).addScore();
+            if(disability(person,i))
                 list.get(i).addScore();
-            }
-        }
-        //comparing shelters among each other to get the top 3
-        for(int i = 0; i < top3.length; i++){
-            if(list.get(i).getScore() > 0)
-                top3[i] = list.get(i);
         }
+        top3 = sortTop3();
```

Comment:
What's the value in having this as a separate function? Also if you're making it separate, consider passing in the data it needs rather than relying on the object fields to reduce potential bugs. (Ask me later if you want a more detailed explanation for why one might be better than the other.)